### PR TITLE
WIP: Plumbing

### PR DESCRIPTION
**Pull request by datamel**
Draft PR with some inital basic plumbing for platforms.

### CODE REVIEW

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ cylc/flow/cfgspec/globalcfg.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4876113&bid=16161542)

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ cylc/flow/cfgspec/globalcfg.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4876113&bid=16161542)

File: cylc/flow/suite_files.py [hunk 1]
```diff
@@ -242,7 +242,7 @@ def detect_old_contact_file(reg, check_host_port=None):
     cmd = ["timeout", "10", "ps", PS_OPTS, str(old_pid_str)]
     if is_remote_host(old_host):
         import shlex
-        ssh_str = str(glbl_cfg().get_host_item("ssh command", old_host))
```
[the remark this answers is not in the document]
Comment (dpmatthews) [reply cylc/flow/suite_files.py]: Not sure what you mean. `old_host` is where the old contact files claims the suite is running. The code needs to ssh to that host to see if it really is.

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4876113&bid=16161542)

**Comment by hjoliver** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cylc/cylc-flow/pullRequest?prid=4876113&bid=16161542)

**Pull request closed without merging**